addons/binding/org.openhab.binding.silvercrestwifisocket/src/main/java/org/openhab/binding/silvercrestwifisocket/handler/SilvercrestWifiSocketHandler.java

```diff
@@ -258,7 +262,27 @@ private void saveMacAddressFromConfiguration(final Configuration configuration)
     }
 
     /**
-     * Sends one command to the Wifi Socket. If the host address is not setted, it will trigger the lookup of the
+     * Saves the vendor from configuration in field.
+     *
+     * @param configuration The {@link Configuration}
+     */
+    private void saveVendorFromConfiguration(final Configuration configuration) {
+        // TODO: Validation of Vendor!?
+        if ((configuration != null) && (configuration.get(SilvercrestWifiSocketBindingConstants.VENDOR_ARG) != null)) {
+            this.vendor =
+                    // SilvercrestWifiSocketVendor.valueOf(String.valueOf(
+                    SilvercrestWifiSocketVendor.valueOf(
+                            String.valueOf(configuration.get(SilvercrestWifiSocketBindingConstants.VENDOR_ARG)));
+        }
+        /*
+         * if (this.vendor == null) {
```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?

Reply:
Will remove it...